source/data-api/generated-endpoints.txt

```diff
@@ -0,0 +1,1364 @@
+.. _data-api-endpoints:
+
+==================
+Data API Endpoints
+==================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+The Data API lets you securely read and write data using standard HTTPS
+requests. The API includes automatically generated endpoints that each
+represent a MongoDB operation. You can use the endpoints to create,
+read, update, delete, and aggregate documents in a MongoDB data source.
+
+For example, this :mdn:`POST <Web/HTTP/Methods/POST>` request stores a
+document in a linked cluster by calling the ``insertOne`` endpoint:
+
+.. io-code-block::
+   
+   .. input::
+      :language: bash
+      
+      curl --request POST \
+        'https://data.mongodb-api.com/app/myapp-abcde/endpoint/data/v1/action/insertOne' \
+        --header 'Content-Type: application/json' \
+        --header 'Accept: application/json' \
+        --header 'api-key: TpqAKQgvhZE4r6AOzpVydJ9a3tB1BLMrgDzLlBLbihKNDzSJWTAHMVbsMoIOpnM6' \
+        --data-raw '{
+            "dataSource": "mongodb-atlas",
+            "database": "learn-data-api",
+            "collection": "hello",
+            "document": {
+              "text": "Hello from the Data API!",
+            }
+        }'
+   
+   .. output::
+      :language: json
+      
+      { "insertedId": "5f1a785e1536b6e6992fd588" }
+
+When to Use the Data API
+------------------------
+
+You can use the Data API to integrate with any app or service that
+supports HTTPS requests. For example, you might:
+
+- call the API from a serverless edge function
+- access test data and log events in a CI/CD workflow
+- integrate Atlas into a federated API gateway
+- connect from an environment not currently supported via a MongoDB
+  Driver or Realm SDK
+
+An operation called through an API endpoint will likely take longer than
+the corresponding MongoDB operation called through a connected MongoDB
+Driver. For high-load use-cases and latency sensitive applications, we
+recommend connecting directly to your database with a MongoDB driver. To
+learn more, visit the :driver:`MongoDB Drivers </>` documentation.
```

Comment:
also worth calling out that you shouldn't call data api directly from a client b/c of limited auth permissions